### PR TITLE
issue 262 partial check and

### PR DESCRIPTION
**Pull request by murari85b7**
1.internet connection check for overlap button
2.issue #262 partially check(if the app is open and user visited  news section once later if internet not there then also he can see the cache webview )

### CODE REVIEW

File: app/views/LocationTracking.js [hunk 1]
```diff
@@ -36,30 +42,91 @@ class LocationTracking extends Component {
...
-          });
-          this.willParticipate();
-        } else {
-          this.setState({
-            isLogging: false,
-          });
-        }
-      })
-      .catch(error => console.log(error));
+        .then(isParticipating => {
```
Comment (tstirrat) [inline app/views/LocationTracking.js:53]: Can you please ensure your Prettier formatter is running on save?

The indent set in prettier is 2 spaces.

```
tabWidth: 2,
useTabs: false,
```

File: app/views/LocationTracking.js [hunk 2]
```diff
@@ -36,30 +42,91 @@ class LocationTracking extends Component {
...
+    let timer_intersect = setInterval(this.intersect_tick, 1000 * 60 * 60 * 12); // once every 12 hours
+    // DEBUG:  1000 * 10); // once every 10 seconds
+
+    this.setState({
+      timer_intersect,
+    });
   }
+
+  intersect_tick = () => {
+    // This function is called once every 12 hours.  It should do several things:
```
Comment (tstirrat) [inline app/views/LocationTracking.js:76]: * If the function needs an outline/table of contents, please split it into smaller functions
* Use async await

```js
const authorities = await getUserAuthorities();
if (authority_list) {
  await fetchAuthorities(authority_list);
}
```

File: app/views/LocationTracking.js [hunk 3]
```diff
@@ -36,30 +42,91 @@ class LocationTracking extends Component {
...
+            this.willParticipate();
+          } else {
+            this.setState({
+              isLogging: false,
+            });
+          }
+        })
+        .catch(error => console.log(error));
+
+    let timer_intersect = setInterval(this.intersect_tick, 1000 * 60 * 60 * 12); // once every 12 hours
```
Comment (tstirrat) [inline app/views/LocationTracking.js:67]: setting long timeouts and intervals is not recommended. We should use background task aware libs like:  https://www.npmjs.com/package/react-native-background-timer

File: app/views/LocationTracking.js [hunk 4]
```diff
@@ -76,7 +143,13 @@ class LocationTracking extends Component {
   }
 
   overlap() {
-    this.props.navigation.navigate('OverlapScreen', {});
+    NetInfo.fetch().then(state => {
+      if(state.isConnected)
+        this.props.navigation.navigate('OverlapScreen', {});
+      else
+        alert('Please check your internet connection and try again.')
```
Comment (tstirrat) [inline app/views/LocationTracking.js:150]: This string will need to be translated

File: app/views/LocationTracking.js [hunk 5]
```diff
@@ -111,6 +184,21 @@ class LocationTracking extends Component {
     this.props.navigation.navigate('LicensesScreen', {});
   }
 
+  settings() {
+    this.props.navigation.navigate('SettingsScreen', {});
+  }
+
+  willParticipate = () => {
```
Comment (tstirrat) [inline app/views/LocationTracking.js:191]: duplicate willParticipate, this is possible a merge conflict.. please sync, and remove if still there

File: app/views/LocationTracking.js (hunk 3 above)
Comment (penrods) [reply app/views/LocationTracking.js:67]: I agree, but this wasn't introduced by the PR.  Not really this contributor's responsibility.

File: app/views/LocationTracking.js (hunk 2 above)
Comment (penrods) [reply app/views/LocationTracking.js:76]: I agree, but this wasn't introduced by the PR.  Not really this contributor's responsibility.

File: app/views/LocationTracking.js (hunk 4 above)
Comment (murari85b7) [reply app/views/LocationTracking.js]: Fixed this issue

Comment (murari85b7) [reply app/views/LocationTracking.js]: Fixed this issue

File: app/views/LocationTracking.js (hunk 3 above)
Comment (murari85b7) [reply app/views/LocationTracking.js]: this is not changed by me. May be an issue with some space

File: app/views/LocationTracking.js (hunk 2 above)
Comment (murari85b7) [reply app/views/LocationTracking.js]: This is not changed by me. May be an issue with some space.

File: app/views/LocationTracking.js (hunk 1 above)
Comment (murari85b7) [reply app/views/LocationTracking.js]: This is not changed by me. May be an issue with some space.

File: app/views/LocationTracking.js (hunk 5 above)
Comment (murari85b7) [reply app/views/LocationTracking.js]: This is not changed by me. Maybe an issue with some space.

**Comment by murari85b7** [comment]
I have changed based on the above comment. Can you please verify and merge it?

File: app/views/LocationTracking.js (hunk 3 above)
Comment (tstirrat) [reply app/views/LocationTracking.js]: Oops, my bad. I got confused by the branch not being up to date.

**Comment by tstirrat** [comment]
Sorry for the confusion, I was not able to see what was this PR and what was branch staleness.

There are still some merge conflicts, can you merge develop and try to resolve them?


Also, there was a large UI refresh (#354 ), so there might need to be a conceptual change to this PR. The check overlap button is gone, however there is an event history button in settings screen now.

**Comment by murari85b7** [comment]
Thanks for the confirmation. I can see the News and Loactiontracker js file change. But I have one question.
Do we have still #262 issues valid?

**Comment by tstirrat** [comment]
Hi, yes #262 is still valid. There is a news section in **Settings screen**. I would recommend splitting that functionality out of this PR and tackling it in another PR.

There might be more than one health authority active, so it needs more thought

**Comment by murari85b7** [comment]
Thanks @tstirrat . Sure I will create new PR

**Pull request closed without merging**